src/articles/components/ArticleView/RelatedArticles/index.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { FC, useEffect, useState } from 'react';
+import { Link } from 'react-router-dom';
+import { DOMAIN } from '../../../../common/constants/endpoints';
+import { getArticles } from '../../../api';
+import { IArticle } from '../../../models/Article';
+import { routes } from '../../ArticlesRouter';
+
+import style from '../articleView.less';
+
+export interface IProps {
+  mainArticle: IArticle;
+}
+
+export const RelatedArticles: FC<IProps> = ({ mainArticle }) => {
+  const [relatedArticles, setRelatedArticles] = useState<IArticle[]>([]);
+
+  const updateRelatedArticleList = (articles: IArticle[]) => {
+    const oldArticles = relatedArticles.filter((oldArticle) => {
+      const newArticleIds = articles.map((article) => article.id);
+      return !newArticleIds.includes(oldArticle.id);
+    });
+    const allArticles = oldArticles
+      .concat(articles)
+      .filter((article) => article.id !== mainArticle.id)
+      .filter((article) => article.tags.some((articleTag) => mainArticle.tags.includes(articleTag)));
+
+    return allArticles
+      .filter((article, idx, final) => final.map((e) => e.id).indexOf(article.id) === idx)
+      .sort((a, b) => {
+        return Date.parse(b.published_date) - Date.parse(a.published_date);
+      });
+  };
+
+  useEffect(() => {
+    const fetchArticles = async () => {
+      let incomingRelatedArticleList: IArticle[] = [];
+      for (const tag of mainArticle.tags) {
+        const tagArticles = await getArticles({ tags: tag });
+        incomingRelatedArticleList = incomingRelatedArticleList.concat(tagArticles);
+      }
```

Comment:
I have implementet your `Promise.all()`, but we cannot integrate it with the OW4 API since if you send a list of tags there you will get only the articles with **all** the tags on it: [Example](https://online.ntnu.no/api/v1/articles/?tags=%C3%B8konomi&tags=fondstyret)

Reply:
I think we could at the very least consider modifying the filter, or adding a new one for filtering on multiple tags. Like a `tags__in` or something?

Not really a need for it currently though, this work fine.

Reply:
I think the current solution is fine, but I added an issue for the filter we want: https://github.com/dotkom/onlineweb4/issues/2238, so that someone can add it at a later point